Amplify/Categories/Auth/Error/AuthError.swift

```diff
@@ -14,6 +14,8 @@ public enum AuthError {
     case validation(Field, ErrorDescription, RecoverySuggestion, Error? = nil)
     case notAuthorized(ErrorDescription, RecoverySuggestion, Error? = nil)
     case invalidState(ErrorDescription, RecoverySuggestion, Error? = nil)
+    case signedOut(ErrorDescription, RecoverySuggestion, Error? = nil)
+    case sessionExpired(ErrorDescription, RecoverySuggestion, Error? = nil)
```

Comment:
1. Add doc comments for all cases
1. Are these error cases common across platforms?